src/core_plugins/kibana/public/dashboard/get_top_nav_config.js

```diff
@@ -0,0 +1,84 @@
+import { DashboardViewMode } from './dashboard_view_mode';
+import { createTopNavExecuteConfig, createTopNavTemplateConfig } from 'ui/kbn_top_nav/kbn_top_nav_config';
+
+/**
+ * @param {DashboardMode} dashboardMode.
+ * @param kbnUrl - used to change the url.
+ * @param {(DashboardMode) => void} modeChange - a function to trigger a dashboard mode change.
+ * @return {Array} - Returns an array of objects for a top nav configuration, based on the
+ * mode.
+ */
+export function getTopNavConfig(dashboardMode, kbnUrl, modeChange) {
+  switch (dashboardMode) {
+    case DashboardViewMode.VIEW:
+      return [getNewConfig(kbnUrl), getOpenConfig(), getShareConfig(), getEditConfig(modeChange)];
+    case DashboardViewMode.EDIT:
+      return [getNewConfig(kbnUrl), getOpenConfig(), getAddConfig(), getSaveConfig(), getOptionsConfig(), getViewConfig(modeChange)];
+    default:
+      return [];
+  }
+}
+
+function getEditConfig(modeChange) {
+  return createTopNavExecuteConfig(
+    'edit',
+    'Switch to edit mode',
+    'dashboardEditMode',
+    () => { modeChange(DashboardViewMode.EDIT); });
+}
+
+function getViewConfig(modeChange) {
+  return createTopNavExecuteConfig(
+    'Stop Editing',
```

Comment:
because of the case?  I did that on purpose because I want the second word capitalized.  The first word gets auto capitalized but the second doesn't.  I could do `stop editing` and it'll show up as `Stop editing` but I didn't like that.  I could do `stop Editing` but that's weird too.  I suppose I could also find where these words are capitalized and capitalize every word there, but that might affect other tabs.  I'll revert to `stop editing` to be consistent here and we can decide on the casing later, and change it for all or nothing.